site/app/models/Button.php

```diff
@@ -38,6 +38,9 @@ class Button extends AbstractModel {
     protected $title;
     /** @prop @var string|null $subtitle */
     protected $subtitle;
+    /** @prop @var DateTime|null A DateTime object representing the time for this Button.  For example if this was a
+    * grade button then $date may represent the date and time grading begins or ends */
+    protected $date;
```

Comment:
add `@method getDate(): DateTime|null` above. Also add test for function.

Reply:
Done.